utils/modules_inspection.py

```diff
@@ -32,17 +33,19 @@ def iter_classes_from_module(module, base_types):
                     yield cls
 
 
-def iter_submodule_names(path, root=""):
+def iter_submodule_names(path, depth=0, root="", current_depth=1):
     """
     Return all sub modules of given module (all sub folders of given folder)
     :param path: absolute path - str
+    :param depth: 0 means it returns all submodules, if 1 it returns only modules from given directory, if 2 ... so on
     :param root: for internal work
+    :param current_depth: for internal work
     :return: module name - str
     """
     for _, module_name, is_package in pkgutil.iter_modules([str(path)]):
-        if is_package:
-            sub_path = path / module_name
+        if is_package and (depth == 0 or current_depth < depth):
+            sub_path = f'{path} // {module_name}' 
```

Comment:
is path variable there a string or a Path object?

Reply:
why was it ever `path / module_name` ? : )  wow.

Reply:
It got path object before but I decided to change it to string.

Reply:
and `//`  will work on all operating systems? or should it be `os.path.join(path, module_name)` to be sure ?

Reply:
Did not know. I will think.